lib/reader.js

```diff
@@ -603,13 +601,12 @@ class ParquetEnvelopeReader {
     if (colChunk.meta_data.dictionary_page_offset) {
       const offset = +colChunk.meta_data.dictionary_page_offset;
       const size = Math.min(+this.fileSize - offset, this.default_dictionary_size);
-      dictionary = this.read(offset, size, colChunk.file_path).then(buffer => decodePage({offset: 0, buffer, size: buffer.length}, opts).dictionary);
+      const buffer = await this.read(offset, size, colChunk.file_path)
+      const dict = await decodePage({offset: 0, buffer, size: buffer.length}, opts)
```

Comment:
Is is necessary to read and decode the dictionary if opts.dictionary is already set? It'd be nice to avoid it if we're just going to throw it away, but maybe it's required to start the next read in the right place.

Reply:
It's not absolutely required, it just made it a little more readable.